src/net_processing.cpp

```diff
@@ -2806,6 +2820,61 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         return;
     }
 
+    // Received from an inbound peer planning to reconcile transactions with us, or
+    // from an outgoing peer demonstrating readiness to do reconciliations.
+    // If received from outgoing, adds the peer to the reconciliation queue.
+    // Feature negotiation of tx reconciliation should happen between VERSION and
+    // VERACK, to avoid relay problems from switching after a connection is up.
+    if (msg_type == NetMsgType::SENDRECON) {
+        if (!pfrom.m_tx_relay) return;
+        LOCK(peer->m_recon_state_mutex);
+        if (peer->m_recon_state != nullptr) return; // Do not support reconciliation salt/version updates.
+        LOCK(cs_main);
+        if (!State(pfrom.GetId())->m_wtxid_relay) return; // SENDRECON is allowed only after WTXIDRELAY.
+
+        bool recon_requestor, recon_responder;
+        uint64_t remote_salt;
+        uint32_t recon_version;
+        vRecv >> recon_requestor >> recon_responder >> recon_version >> remote_salt;
+        if (recon_version != 1) return;
+
+        // Do not flood through inbound connections which support reconciliation to save bandwidth.
+        bool flood_to = false;
+        if (pfrom.IsInboundConn()) {
+            // We currently don't support reconciliations with inbound peers which
+            // don't want to be reconciliation senders (request our sketches),
+            // or want to be reconciliation responders (send us their sketches).
+            // Just ignore SENDRECON and use normal flooding for transaction relay with them.
+            if (!recon_requestor) return;
+            if (recon_responder) return;
+        } else {
+            // We currently don't support reconciliations with outbound peers which
+            // don't want to be reconciliation responders (send us their sketches),
+            // or want to be reconciliation senders (request our sketches).
+            // Just ignore SENDRECON and use normal flooding for transaction relay with them.
+            if (recon_requestor) return;
+            if (!recon_responder) return;
+            // TODO: Flood only through a limited number of outbound connections.
+            flood_to = true;
+        }
+
+        uint64_t local_salt = peer->m_local_recon_salt;
```

Comment:
this salt calculation (and probably also the flood_to logic) seems like an example of code that can be extracted into the reconciliation module rather than added to `ProcessMessage`.

it would make sense to me for the flow to be something like: 
- `ProcessMessage` does internal state checks (`m_tx_relay` exists, wtxid is enabled)
- extract the message from `vRecv`
- pass it along into the module to handle 
- module returns saying "no-op" or "here is a ReconState object" 
- if it exists, `ProcessMessage` stores the object on the peer

I suspect we could go even further, but this would be a start. Ideally the module would abstract this nuanced erlay specific logic away from net processing, even if its per-peer. The txrequest module is a good example for this, eg. `PeerInfo` struct and how its applied within the module to make choices, and the decisions are surfaced to `net_processing`.

Reply:
Yeah, I'm planning to do something similar in the next couple days. That's why I marked the PR draft for now :)

Thank you for the design suggestion!